File: kotlin/src/main/java/LongestCommonPrefix.java
public class LongestCommonPrefix {

    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) return "";

        String first = strs[0];
        String current;
        int index = first.length();
        for (int i = 0; i < first.length(); i++) {
            for (int j = 1; j < strs.length; j++) {
                current = strs[j];
                if (i >= current.length() || first.charAt(i) != current.charAt(i)) {
                    return first.substring(0,i);
                }
            }
            index = i;
        }
        return first.substring(0,index);
    }
}

File: kotlin/src/main/kotlin/LongestCommonPrefix.kt
package org.wyekings

class LongestCommonPrefix {
    fun longestCommonPrefix1(strs: Array<String>): String {
        if (strs.isEmpty()) return ""
        var prefix = ""
        val first = strs[0]
        first.forEachIndexed { index, c ->
            for (i in 1..<strs.size) {
                val current = strs[i]
                if (index < current.length && c != current[index]) return prefix
            }
            prefix += first[index]
        }

        return prefix
    }
}

fun main() {
    val solution = LongestCommonPrefix()
    val strs = arrayOf("flower","flow","flight")
    val result = solution.longestCommonPrefix1(strs)
    println("result=$result")

}